hathor/p2p/peer_id.py

```diff
@@ -83,6 +83,9 @@ def __str__(self):
         return ('PeerId(id=%s, entrypoints=%s, retry_timestamp=%d, retry_interval=%d)' % (self.id, self.entrypoints,
                 self.retry_timestamp, self.retry_interval))
 
+    def __hash__(self):
```

Comment:
Maybe add a comment explaining that two different objects with same id will be considered the same?

Reply:
I need to test this better, it's certainly more complex than that, `__hash__` doesn't cause object equality. Also this change might be wrong, it seems `__hash__` _requires_ object equality in the first place.

Reply:
So I feel we should take it out of this PR. What do you think?

Reply:
I'll remove this along with `sorted/Set` changes.

Reply:
Removed.